schemaregistry/resource_schema.go

```diff
@@ -23,13 +24,33 @@ func resourceSchema() *schema.Resource {
 			StateContext: schema.ImportStatePassthroughContext,
 		},
 		CustomizeDiff: customdiff.ComputedIf("version", func(ctx context.Context, d *schema.ResourceDiff, meta interface{}) bool {
+
+			var schemaHasChange bool
 			oldState, newState := d.GetChange("schema")
-			newJSON, _ := structure.NormalizeJsonString(newState)
-			oldJSON, _ := structure.NormalizeJsonString(oldState)
-			schemaHasChange := newJSON != oldJSON
 
-			// explicitly set a version change on schema change and make dependencies aware of a
-			// version changed at `plan` time (computed field)
+			if schemaTypeStr, ok := d.Get("schema_type").(string); ok {
+				if strings.ToLower(schemaTypeStr) == "json" {
```

Comment:
Nit. 32-49 is the same as 71-88. Should this be its own function? Easier to unit test.